src/sidenav/sidenav-item.tsx

```diff
@@ -195,6 +199,9 @@ export const SidenavItem = ({
                 {...otherProps}
                 $highlight={isSelected || isCurrent}
                 {...getReferenceProps()}
+                onMouseLeave={() => {
+                    handleMouseLeave(isOpen);
+                }}
```

Comment:
`isOpen` comes from state? could the function directly access the variable instead?

```suggestion
                onMouseLeave={handleMouseLeave}
```